lib/stimulus_reflex.rb

```diff
@@ -18,6 +19,7 @@
 require "stimulus_reflex/broadcasters/page_broadcaster"
 require "stimulus_reflex/broadcasters/selector_broadcaster"
 require "generators/stimulus_reflex_generator"
+require "generators/config_generator"
```

Comment:
This line and the line above it don't actually need to be here and doesn't need to be required into the application if you restructure the directories a little bit. They will be autoloaded if you place them in `lib/generators/stimulus_reflex/` instead of `lib/generators/`. You could then delete these two `require` statements. This would let Rails do its thing when it comes to finding generators on its own as well as avoid unnecessarily adding more classes and such into the actual running Rails app.

Reply:
Oh, nice, thx, I will incorporate that.

Reply:
@dark-panda I tried to do it but I ran into a limitation with that approach: As we the commands to be
* `rails generate stimulus_reflex example` and
* `rails generate stimulus_reflex:config`

For the magic to work, the reflex generator has to be `StimulusReflexGenerator`
and the config generator has to be `StimulusReflex::ConfigGenerator` (as far as I understand)

So to make it work, it expects the corresponding template files to be corresponding folders, which means some template files will be in `lib/generators/stimulus_reflex/config` but the others in `lib/generators/` (one module-namespaced under `/stimulus_reflex` the other not)

I find that somewhat confusing, to have to search for the templates in two places.

* Do you agree that having two template folders is confusing, or is that a better practice, because Rails wants it that way?
* Or is there another way to do this? Any suggestions on what would be best to do here?

Reply:
Idea from @ParamagicDev just create some rake tasks, then the commands can be independent of the internal structure. 

> Is the goal just to add rake tasks keep the CLI the same? Because you could easily create a tasks folder that holds both and you can define the commands there's in a stimulus_reflex.rake file. 

On the other hand, are we bikeshedding here? (Genuine question)

Reply:
@RolandStuder It seemed to work okay for me without any special renaming. Here's my directory structure:

```
lib/generators/stimulus_reflex/config_generator.rb
lib/generators/stimulus_reflex/templates/app/reflexes/application_reflex.rb.tt
lib/generators/stimulus_reflex/templates/app/reflexes/%file_name%_reflex.rb.tt
lib/generators/stimulus_reflex/templates/app/javascript/controllers/%file_name%_controller.js.tt
lib/generators/stimulus_reflex/templates/app/javascript/controllers/application_controller.js.tt
lib/generators/stimulus_reflex/templates/config/initializers/stimulus_reflex.rb
lib/generators/stimulus_reflex/stimulus_reflex_generator.rb
```

All of the templates generated fine and I didn't touch the code in any files at all other than to remove the `requires`.

Reply:
@dark-panda Yeah, you are right, must have messed up something. Now everything is in autoloaded paths and I was able to remove the requires. Thx for your help.

Reply:
@dark-panda Hm, but now I had to require the Generator in the Testfile, I tried to move the testfile as well into `test/generators/stimulus_reflex/stimulus_reflex_generator_test.rb` but with no success (I get `NameError: uninitialized constant StimulusReflex::StimulusReflexGenerator` or if I adjust `test ::StimulusReflexGenerator` I get `NameError: uninitialized constant StimulusReflexGenerator` unless I require the generator in the test file.

Autoloading is somewhat obscure to me. Happy to accept a PR towards this branch ;-)

Reply:
Not a problem, I’ll write one up tomorrow.

Reply:
Hey gang, I'm doing my best to follow along but autoloading is black magic to me.

I'm hovering around the merge button because this PR unlocks us moving forward with several others (including yours, @dark-panda!) but it's not clear if the Testfile/Generator stuff is actually blocking. If I merge this, is the PR you're discussing something that could be made against the master branch instead?

I don't want to rush anyone, either. If you tell me to chill out, I will. 😁